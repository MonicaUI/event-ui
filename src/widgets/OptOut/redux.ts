/* eslint-disable import/no-cycle */

import { injectPageSnapshot } from '../../redux/actions';
import querystring from 'querystring';

/**
 * Opt-out and opt-in Redux module:
 * Event invitees can click "Opt Out" link in their emails to opt them out all communications from the event account
 */
const GET_OPT_OUT_STATUS = 'event-guestside-site/optOut/GET_OPT_OUT_STATUS';
const OPT_OUT_CONTACT = 'event-guestside-site/optOut/OPT_OUT_CONTACT';

const createGetOptedOutStatusAction = contactInfo => {
  return {
    type: GET_OPT_OUT_STATUS,
    payload: contactInfo
  };
};

export const getOptOutStatus =
  (): $TSFixMe =>
  async (dispatch: $TSFixMe, getState: $TSFixMe): Promise<$TSFixMe> => {
    const {
      clients: { eventGuestClient },
      accessToken,
      event: { id: eventId }
    } = getState();
    let {
      userSession: { inviteeId }
    } = getState();
    // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
    const queryParams = window && window.location && querystring.parse(window.location.search.slice(1));
    const guestId = queryParams.g;
    const contactId = queryParams.c;
    if (contactId) {
      inviteeId = null;
    }
    const optedOutContact = await eventGuestClient.getOptOutStatus(accessToken, eventId, inviteeId, guestId, contactId);
    dispatch(createGetOptedOutStatusAction(optedOutContact));
  };

export const createOptedOutAction = (contactInfo: $TSFixMe): $TSFixMe => {
  return {
    type: OPT_OUT_CONTACT,
    payload: contactInfo
  };
};

export const optOutAction =
  (optOutStatus: $TSFixMe, fromConfirmPage: $TSFixMe): $TSFixMe =>
  async (dispatch: $TSFixMe, getState: $TSFixMe): Promise<$TSFixMe> => {
    const {
      clients: { eventGuestClient },
      accessToken,
      event: { id: eventId }
    } = getState();
    let {
      userSession: { inviteeId }
    } = getState();
    // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
    const queryParams = window && window.location && querystring.parse(window.location.search.slice(1));
    const guestId = queryParams.g;
    const contactId = queryParams.c;
    if (contactId) {
      inviteeId = null;
    }
    // eslint-disable-next-line import/no-cycle
    const page = (await import(/* webpackChunkName optOut */ './content')).optInOutPageNew;
    if ((fromConfirmPage && optOutStatus) || !fromConfirmPage) {
      const optedOutContact = await eventGuestClient.optOut(
        accessToken,
        eventId,
        inviteeId,
        guestId,
        contactId,
        optOutStatus
      );
      dispatch(createOptedOutAction(optedOutContact));
    }
    dispatch(injectPageSnapshot(page));
  };

export const prepareForOptOutPageLoad =
  (): $TSFixMe =>
  async (dispatch: $TSFixMe, getState: $TSFixMe): Promise<$TSFixMe> => {
    const {
      userSession: { inviteeId }
    } = getState();
    // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
    const queryParams = window && window.location && querystring.parse(window.location.search.slice(1));
    const guestId = queryParams.g;
    const contactId = queryParams.c;
    let page;
    if (!inviteeId && !guestId && !contactId) {
      page = (await import(/* webpackChunkName optOut */ './invalidInviteeContent')).invalidInviteeOptOut;
    } else {
      await dispatch(getOptOutStatus());
      if (getState() && getState().optOut && getState().optOut.optOutStatus) {
        // eslint-disable-next-line import/no-cycle
        page = (await import(/* webpackChunkName optOut */ './content')).optInOutPage;
      } else {
        // eslint-disable-next-line import/no-cycle
        page = (await import(/* webpackChunkName optOut */ './content')).confirmationPage;
      }
    }
    dispatch(injectPageSnapshot(page));
  };

export const optOutReducer = (state = {}, action: $TSFixMe): $TSFixMe => {
  switch (action.type) {
    case GET_OPT_OUT_STATUS:
    case OPT_OUT_CONTACT:
      return {
        ...state,
        ...action.payload
      };
    default:
      return state;
  }
};
