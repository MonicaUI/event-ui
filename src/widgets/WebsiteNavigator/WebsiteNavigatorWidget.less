.skipNav {
    left: -999px;
    position: absolute;
    top: auto;
    width: 1px;
    height: 1px;
    overflow: hidden;
    z-index: -999;
    float: left;
    &:focus, &:active {
    left: auto;
    padding-left: 2.5px;
    padding-right: 2.5px;
    margin-top: 2.5px;
    position: absolute;
    overflow: auto;
    width: auto;
    height: auto;
    border-radius: 5px;
    border: 2px solid;
    text-align: center;
    z-index: 999;
    }
}