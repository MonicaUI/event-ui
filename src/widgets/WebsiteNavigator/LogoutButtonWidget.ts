import { connect } from 'react-redux';
import ContainerlessButtonWidget from 'nucleus-widgets/lib/Button/ContainerlessButtonWidget';
import { routeToUrl, getPagePath, getCurrentPageId } from '../../redux/pathInfo';
import { getDefaultWebsitePageId } from '../../redux/website';
import { logoutRegistrant } from '../../redux/registrantLogin/actions';
import {
  showLoadingDialog,
  hideLoadingDialog,
  hideLoadingOnError
} from 'nucleus-guestside-site/src/redux/modules/dialogContainer';
import { defaultRegistrationTypeId } from 'event-widgets/utils/registrationType';
import qs from 'querystring';
import {
  REGISTRATION,
  APPROVAL_DENIED,
  CANCELLATION,
  GUEST_REGISTRATION,
  PENDING_APPROVAL,
  POST_REGISTRATION,
  POST_REGISTRATION_PAYMENT
} from '../../redux/website/registrationProcesses';
import { filterEventSnapshot } from '../../redux/actions';
import { getDefaultRegistrationPath } from 'event-widgets/redux/selectors/appData';
import { invalidateDatatagCache } from '../../utils/datatagUtils';

export const stripInviteeFromUrl = (url: $TSFixMe): $TSFixMe => {
  const path = url.split('?')[0];
  const query = qs.parse(url.split('?')[1]) || {};
  delete query.inviteeId;
  // The i query param is the incoded version of the inviteeId.
  delete query.i;
  // The tm query param store a hash for test mode
  delete query.tm;

  // we need to delete marketo and eloqua param from url on log out
  delete query.MarketoID;
  delete query.EloquaID;

  return path + (Object.keys(query).length ? `?${qs.stringify(query)}` : '');
};

function currentPageNeedsLoggedInUser(state) {
  return (
    REGISTRATION.isTypeOfCurrentPage(state) ||
    GUEST_REGISTRATION.isTypeOfCurrentPage(state) ||
    POST_REGISTRATION.isTypeOfCurrentPage(state) ||
    CANCELLATION.isTypeOfCurrentPage(state) ||
    PENDING_APPROVAL.isTypeOfCurrentPage(state) ||
    APPROVAL_DENIED.isTypeOfCurrentPage(state) ||
    POST_REGISTRATION_PAYMENT.isTypeOfCurrentPage(state)
  );
}

function logout() {
  return async (dispatch: $TSFixMe, getState: $TSFixMe): Promise<$TSFixMe> => {
    dispatch(showLoadingDialog());
    let state = getState();
    const { eventSnapshotVersion, appData } = state;
    const defaultRegPathId = getDefaultRegistrationPath(appData);
    try {
      const isInviteeRedirectedToOktaLogoutUrl = await dispatch(logoutRegistrant());
      // after logout is complete, reset website to default regTypeId and default regPathId
      await dispatch(filterEventSnapshot(eventSnapshotVersion, defaultRegistrationTypeId, defaultRegPathId));
      state = getState();
      const nextPageId = currentPageNeedsLoggedInUser(state) ? getDefaultWebsitePageId(state) : getCurrentPageId(state);
      if (!isInviteeRedirectedToOktaLogoutUrl) {
        if (state.defaultUserSession.isTestMode) {
          // setting href instead of using replace so that back button works correctly
          global.location.href = stripInviteeFromUrl(getPagePath(state, nextPageId));
        } else {
          await dispatch(routeToUrl(stripInviteeFromUrl(getPagePath(state, nextPageId))));
        }
      }
    } catch (error) {
      dispatch(hideLoadingOnError());
      throw error;
    }
    dispatch(hideLoadingDialog());
    invalidateDatatagCache();
  };
}

export function mapStateToProps(): $TSFixMe {
  return {
    kind: 'button'
  };
}
export const mapDispatchToProps = {
  clickHandler: logout
};

/**
 * Data wrapper for the log out widget.
 */
export default connect(mapStateToProps, mapDispatchToProps)(ContainerlessButtonWidget);
