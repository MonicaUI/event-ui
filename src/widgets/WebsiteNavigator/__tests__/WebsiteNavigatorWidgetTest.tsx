import { stripInviteeFromUrl } from '../LogoutButtonWidget';

jest.mock('../../../redux/pathInfo', () => {
  return {
    getPagePath: jest.fn(() => 'url?MarketoID=marketo-id'),
    getCurrentPageId: jest.fn(() => 'summary'),
    isInviteeInUrl: jest.fn(() => false),
    routeToPage: jest.fn(),
    routeToUrl: jest.fn(),
    setNavigationDialogConfig: jest.fn()
  };
});
jest.mock('../../../redux/registrantLogin/actions');
jest.mock('../../../redux/selectors/currentRegistrant');
jest.mock('../../../redux/actions');
jest.mock('../../../redux/registrantLogin/actions');

import React from 'react';
import WebsiteNavigatorWidgetWrapper from '../WebsiteNavigatorWidget';
import EventSnapshot from '../../../../fixtures/EventSnapshot.json';
import { Provider } from 'react-redux';
import WebsiteNavigatorWidget from 'nucleus-widgets/lib/WebsiteNavigator/WebsiteNavigatorWidget';
import { filterEventSnapshot } from '../../../redux/actions';
import { getCurrentPageId, getPagePath, routeToPage, routeToUrl } from '../../../redux/pathInfo';
import { logoutRegistrant } from '../../../redux/registrantLogin/actions';
import { isLoggedIn } from '../../../redux/selectors/currentRegistrant';
import EventGuestSideWidgetFactory from '../../../widgetFactory/index';
import { mount } from 'enzyme';
import LogoutButtonWidget from '../LogoutButtonWidget';
import ContainerlessButtonWidget from 'nucleus-widgets/lib/Button/ContainerlessButtonWidget';
import transformEventData from 'event-widgets/utils/transformEventData';
import * as InviteeStatus from 'event-widgets/utils/InviteeStatus';
import { setIn } from 'icepick';

const getState = (isPlanner = false, persona = {}, disablePostRegPageIds = false, isTestMode = false) => {
  const website = EventSnapshot.eventSnapshot.siteEditor.website;
  let eventWebsiteNavigation = website.pluginData.eventWebsiteNavigation;
  if (disablePostRegPageIds) {
    eventWebsiteNavigation = {
      defaultPageId: 'summary',
      // @ts-expect-error ts-migrate(2741) FIXME: Property '"postRegPages"' is missing in type '{}' ... Remove this comment to see the full error message
      navigationGroups: {},
      childIds: ['summary'],
      disabledPageIds: [],
      webLinkOnlyPageIds: []
    };
  }
  const updatedWebsite = setIn(website, ['pluginData', 'eventWebsiteNavigation'], eventWebsiteNavigation);
  return {
    website: updatedWebsite,
    event: EventSnapshot.eventSnapshot,
    appData: transformEventData(
      EventSnapshot.eventSnapshot.siteEditor.eventData,
      EventSnapshot.accountSnapshot,
      EventSnapshot.eventSnapshot,
      EventSnapshot.eventSnapshot.siteEditor.website
    ),
    pathInfo: {
      currentPageId: 'summary',
      rootPath: 'url'
    },
    widgetFactory: {
      loadMetaData: x => (new EventGuestSideWidgetFactory() as $TSFixMe).loadMetaData(x),
      loadComponent: x => (new EventGuestSideWidgetFactory() as $TSFixMe).loadComponent(x)
    },
    text: {
      translate: value => value,
      resolver: {
        fetchAllDataTags: value => value
      }
    },
    userSession: {},
    defaultUserSession: {
      isPlanner,
      isTestMode
    },
    eventSnapshotVersion: 'dummy_event_snapshot_version',
    registrationForm: {
      regCart: {
        eventRegistrations: {
          eventRegId: {
            registrationPathId: 'todoThisShouldBeRegPathId'
          }
        }
      }
    },
    persona
  };
};

const config = {
  shared: {
    pageLinksArrangement: 'HORIZONTAL',
    logo: {
      altText: '',
      asset: {},
      hyperlink: ''
    },
    logoutButton: {
      display: true,
      text: 'WebsiteNavigatorWidget_LogoutButtonText_default__resx'
    },
    registerButton: {
      alreadyRegisteredLink: {
        display: false,
        text: 'WebsiteNavigatorWidget_AlreadyRegisteredLinkText_default__resx'
      },
      display: true,
      text: '_registerNowWidget_textConfig__resx'
    }
  }
};

async function dispatch(action) {
  if (typeof action === 'function') {
    return await action(dispatch, getState);
  }
}
const subscribe = () => {};
const store = { dispatch, subscribe, getState };
const props = {
  config,
  type: 'WebsiteNavigator',
  translate: (resx, options) => (options ? `${resx}:${JSON.stringify(options)}` : resx),
  style: EventSnapshot.eventSnapshot.siteEditor.website.theme.global,
  classes: {},
  id: 'widget:3c9b810e-05ef-444b-a5a9-a0e60b2a961d'
};

const stateWithUserText = {
  ...getState(),
  localizedUserText: {
    currentLocale: 'es-ES',
    localizations: {
      'es-ES': {
        'website.pages.summary.name': 'Summary name'
      }
    }
  }
};

const storeWithUserText = { dispatch, subscribe, getState: () => stateWithUserText };

describe('WebsiteNavigatorWidget', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  test('navigates to page', async () => {
    (isLoggedIn as $TSFixMe).mockImplementation(() => false);
    const wrapper = mount(<WebsiteNavigatorWidgetWrapper {...props} store={store} />);
    const widget = wrapper.find(WebsiteNavigatorWidget);
    await widget.props().navigateToPage('summary');
    expect(routeToPage).toHaveBeenCalledWith('summary');
  });
  test('when user is not logged in it renders', () => {
    (isLoggedIn as $TSFixMe).mockImplementation(() => false);
    const widget = mount(
      // @ts-expect-error ts-migrate(2769) FIXME: No overload matches this call.
      <Provider store={store}>
        <WebsiteNavigatorWidgetWrapper {...props} />
      </Provider>
    );
    expect(widget.find('WebsiteNavigator').first()).toMatchSnapshot();
  });
  test('when user is logged in it renders', () => {
    (isLoggedIn as $TSFixMe).mockImplementation(() => true);
    const widget = mount(
      // @ts-expect-error ts-migrate(2769) FIXME: No overload matches this call.
      <Provider store={store}>
        <WebsiteNavigatorWidgetWrapper {...props} />
      </Provider>
    );
    expect(widget.find('WebsiteNavigator').first()).toMatchSnapshot();
  });
  test('when user is logged in with pending approval status', () => {
    (isLoggedIn as $TSFixMe).mockImplementation(() => true);
    const newStore = {
      dispatch,
      subscribe,
      getState: () =>
        getState(false, {
          inviteeStatus: InviteeStatus.PendingApproval
        })
    };
    const widget = mount(
      // @ts-expect-error ts-migrate(2769) FIXME: No overload matches this call.
      <Provider store={newStore}>
        <WebsiteNavigatorWidgetWrapper {...props} />
      </Provider>
    );
    expect(widget.find('WebsiteNavigator').first()).toMatchSnapshot();
  });
  test('when user is logged in with approval denied status', () => {
    (isLoggedIn as $TSFixMe).mockImplementation(() => true);
    const newStore = {
      dispatch,
      subscribe,
      getState: () =>
        getState(false, {
          inviteeStatus: InviteeStatus.DeniedApproval
        })
    };
    const widget = mount(
      // @ts-expect-error ts-migrate(2769) FIXME: No overload matches this call.
      <Provider store={newStore}>
        <WebsiteNavigatorWidgetWrapper {...props} />
      </Provider>
    );
    expect(widget.find('WebsiteNavigator').first()).toMatchSnapshot();
  });

  async function logout() {
    (isLoggedIn as $TSFixMe).mockImplementation(() => true);
    const component = mount(
      // @ts-expect-error ts-migrate(2769) FIXME: No overload matches this call.
      <Provider store={store}>
        {/* @ts-expect-error ts-migrate(2322) FIXME: Type '{ config: { display: boolean; text: string; ... Remove this comment to see the full error message */}
        <LogoutButtonWidget {...props} config={config.shared.logoutButton} />
      </Provider>
    );
    const widget = component.find(ContainerlessButtonWidget);
    await widget.props().clickHandler(props);
  }

  test('Check if registrant logout is initiated when user clicks on logout button', async () => {
    await logout();
    expect(logoutRegistrant).toHaveBeenCalled();
    expect(routeToUrl).toHaveBeenCalled();
    expect(filterEventSnapshot).toHaveBeenCalled();
  });
  test('Check if registrant logout is initiated when user clicks on logout button and isInviteeRedirectedToOktaLogoutUrl is set to true', async () => {
    (logoutRegistrant as $TSFixMe).mockImplementation(() => () => true);
    await logout();
    expect(logoutRegistrant).toHaveBeenCalled();
    expect(routeToUrl).not.toHaveBeenCalled();
  });
  test('Check if registrant logout is initiated when user clicks on logout button and isInviteeRedirectedToOktaLogoutUrl is set to false', async () => {
    (logoutRegistrant as $TSFixMe).mockImplementation(() => () => false);
    await logout();
    expect(logoutRegistrant).toHaveBeenCalled();
    expect(routeToUrl).toHaveBeenCalled();
  });
  test('Check on registrant logout, defaultPageId for defaultRegType is used', async () => {
    (logoutRegistrant as $TSFixMe).mockImplementation(() => () => false);
    (getCurrentPageId as $TSFixMe).mockImplementation(() => 'confirmation');
    const updatedDefaultPageId = 'summary2';
    (filterEventSnapshot as $TSFixMe).mockImplementation(() => {
      getState().website.pluginData.eventWebsiteNavigation.defaultPageId = updatedDefaultPageId;
    });
    await logout();
    expect(filterEventSnapshot).toHaveBeenCalled();
    expect(getPagePath).toHaveBeenCalledWith(expect.anything(), updatedDefaultPageId);
  });
  test('does not renders when it is planner mode', () => {
    (isLoggedIn as $TSFixMe).mockImplementation(() => false);
    const newStore = { dispatch, subscribe, getState: () => getState(true) };
    const widget = mount(
      // @ts-expect-error ts-migrate(2769) FIXME: No overload matches this call.
      <Provider store={newStore}>
        <WebsiteNavigatorWidgetWrapper {...props} />
      </Provider>
    );
    expect(widget.find('WebsiteNavigator').first()).toMatchSnapshot();
  });
  test(`that it does not fall over with null ptr exceptions when a re-render is caused by login actions,
    but the post reg content hasn't been loaded yet`, () => {
    (isLoggedIn as $TSFixMe).mockImplementation(() => true);
    const newStore = { dispatch, subscribe, getState: () => getState(false, {}, true) };
    const widget = mount(
      // @ts-expect-error ts-migrate(2769) FIXME: No overload matches this call.
      <Provider store={newStore}>
        <WebsiteNavigatorWidgetWrapper {...props} />
      </Provider>
    );
    expect(widget.find('WebsiteNavigator').first()).toMatchSnapshot();
  });
  test('renders when userText has data and currentLocaleId is set', () => {
    (isLoggedIn as $TSFixMe).mockImplementation(() => false);
    (getCurrentPageId as $TSFixMe).mockImplementation(() => 'summary');
    const widget = mount(
      // @ts-expect-error ts-migrate(2769) FIXME: No overload matches this call.
      <Provider store={storeWithUserText}>
        <WebsiteNavigatorWidgetWrapper {...props} />
      </Provider>
    );
    expect(widget.find('WebsiteNavigator').first()).toMatchSnapshot();
  });
  test('Check if marketo query params are removed after logout is initiated', async () => {
    const pathUrl = 'url?MarketoID=marketo-id';
    const strippedURL = 'url';
    expect(stripInviteeFromUrl(pathUrl)).toEqual(strippedURL);
  });
  test('Check if eloqua query params are removed after logout is initiated', async () => {
    const pathUrl = 'url?EloquaID=eloqua-id';
    const strippedURL = 'url';
    expect(stripInviteeFromUrl(pathUrl)).toEqual(strippedURL);
  });
  function registerNowWidgetNotVisibleOnGivenPage(pageId) {
    (isLoggedIn as $TSFixMe).mockImplementation(() => false);
    (getCurrentPageId as $TSFixMe).mockImplementation(() => pageId);
    const newStore = {
      dispatch,
      subscribe,
      getState: () => getState(false, {}, false, true)
    };
    const widget = mount(
      // @ts-expect-error ts-migrate(2769) FIXME: No overload matches this call.
      <Provider store={newStore}>
        <WebsiteNavigatorWidgetWrapper {...props} />
      </Provider>
    );
    expect(widget.containsMatchingElement(<span className="">_registerNowWidget_textConfig__resx</span>)).toBeFalsy();
  }
  test('Register now button is not visible if current page is optInOut in testMode', () => {
    registerNowWidgetNotVisibleOnGivenPage('optInOut');
  });
  test('Register now button is not visible if current page is unsubscribeSubscribe in testMode', () => {
    registerNowWidgetNotVisibleOnGivenPage('unsubscribeSubscribe');
  });
  test('Register now button is not visible in container events if website is off', () => {
    const containerEventStore = {
      dispatch,
      subscribe,
      getState: () => ({
        ...stateWithUserText,
        event: {
          ...stateWithUserText.event,
          isContainerEvent: true,
          containerFeatures: { isWebsiteEnabled: false }
        }
      })
    };
    const widget = mount(
      // @ts-expect-error ts-migrate(2769) FIXME: No overload matches this call.
      <Provider store={containerEventStore}>
        <WebsiteNavigatorWidgetWrapper {...props} />
      </Provider>
    );
    expect(widget.containsMatchingElement(<span className="">_registerNowWidget_textConfig__resx</span>)).toBeFalsy();
  });
  test('Register now button is not visible in container events if registration is off', () => {
    const containerEventStore = {
      dispatch,
      subscribe,
      getState: () => ({
        ...stateWithUserText,
        event: {
          ...stateWithUserText.event,
          isContainerEvent: true,
          containerFeatures: { isRegistrationEnabled: false }
        }
      })
    };
    const widget = mount(
      // @ts-expect-error ts-migrate(2769) FIXME: No overload matches this call.
      <Provider store={containerEventStore}>
        <WebsiteNavigatorWidgetWrapper {...props} />
      </Provider>
    );
    expect(widget.containsMatchingElement(<span className="">_registerNowWidget_textConfig__resx</span>)).toBeFalsy();
  });
  test('Website pages are not visible in container events if website is off', () => {
    const containerEventStore = {
      dispatch,
      subscribe,
      getState: () => ({
        ...stateWithUserText,
        event: {
          ...stateWithUserText.event,
          isContainerEvent: true,
          containerFeatures: { isRegistrationEnabled: true, isWebsiteEnabled: false }
        }
      })
    };
    const widget = mount(
      // @ts-expect-error ts-migrate(2769) FIXME: No overload matches this call.
      <Provider store={containerEventStore}>
        <WebsiteNavigatorWidgetWrapper {...props} />
      </Provider>
    );
    expect(widget.contains('_defaultPageTitle_summary__resx')).toBeFalsy();
  });

  test('Post reg pages are visible in container events even if website is off', () => {
    (isLoggedIn as $TSFixMe).mockImplementation(() => true);
    (getCurrentPageId as $TSFixMe).mockImplementation(() => 'confirmation');
    const containerEventStore = {
      dispatch,
      subscribe,
      getState: () => ({
        ...stateWithUserText,
        event: {
          ...stateWithUserText.event,
          isContainerEvent: true,
          containerFeatures: { isRegistrationEnabled: true, isWebsiteEnabled: false }
        }
      })
    };
    const widget = mount(
      // @ts-expect-error ts-migrate(2769) FIXME: No overload matches this call.
      <Provider store={containerEventStore}>
        <WebsiteNavigatorWidgetWrapper {...props} />
      </Provider>
    );
    expect(widget.contains('_defaultPageTitle_summary__resx')).toBeFalsy();
    expect(widget.contains('_pageManagement_postRegSectionHeader__resx')).toBeTruthy();
  });
});
