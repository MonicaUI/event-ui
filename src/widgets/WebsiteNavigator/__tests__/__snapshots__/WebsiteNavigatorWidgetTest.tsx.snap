// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebsiteNavigatorWidget does not renders when it is planner mode 1`] = `<div />`;

exports[`WebsiteNavigatorWidget renders when userText has data and currentLocaleId is set 1`] = `
<div
  role="navigation"
  style="min-height: 1px;"
>
  <a
    class="skipNav"
    href="#main"
    style="border-color: #041532; color: rgb(39, 63, 105); background-color: rgb(247, 247, 247);"
    tabindex="0"
  >
    WebsiteNavigatorWidget_SkipNavigationText_default__resx
  </a>
  <div
    class=""
    style="padding: 15px 15px 15px 15px; background-color: rgb(4, 21, 50); border-radius: 0; flex: 1 1 auto;"
  >
    <div
      class=""
    >
      <ul
        class=""
      >
        <li
          class=""
        >
          <div
            class=""
            role="button"
            style="text-align: left;"
            tabindex="0"
          >
            Summary name
          </div>
        </li>
      </ul>
    </div>
    <div
      class=""
    >
      <div
        class=""
      >
        <div
          class=""
        >
          <button
            class="disabled"
            disabled=""
            type="button"
          >
            <span
              class=""
            >
              _registerNowWidget_textConfig__resx
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WebsiteNavigatorWidget that it does not fall over with null ptr exceptions when a re-render is caused by login actions,
    but the post reg content hasn't been loaded yet 1`] = `
<div
  role="navigation"
  style="min-height: 1px;"
>
  <a
    class="skipNav"
    href="#main"
    style="border-color: #041532; color: rgb(39, 63, 105); background-color: rgb(247, 247, 247);"
    tabindex="0"
  >
    WebsiteNavigatorWidget_SkipNavigationText_default__resx
  </a>
  <div
    class=""
    style="padding: 15px 15px 15px 15px; background-color: rgb(4, 21, 50); border-radius: 0; flex: 1 1 auto;"
  >
    <div
      class=""
    >
      <ul
        class=""
      >
        <li
          class=""
        >
          <div
            class=""
            role="button"
            style="text-align: left;"
            tabindex="0"
          >
            _defaultPageTitle_summary__resx
          </div>
        </li>
      </ul>
    </div>
    <div
      class=""
    >
      <div
        class=""
      >
        <div
          class=""
        >
          <button
            class=""
            type="button"
          >
            <span
              class=""
            >
              WebsiteNavigatorWidget_LogoutButtonText_default__resx
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WebsiteNavigatorWidget when user is logged in it renders 1`] = `
<div
  role="navigation"
  style="min-height: 1px;"
>
  <a
    class="skipNav"
    href="#main"
    style="border-color: #041532; color: rgb(39, 63, 105); background-color: rgb(247, 247, 247);"
    tabindex="0"
  >
    WebsiteNavigatorWidget_SkipNavigationText_default__resx
  </a>
  <div
    class=""
    style="padding: 15px 15px 15px 15px; background-color: rgb(4, 21, 50); border-radius: 0; flex: 1 1 auto;"
  >
    <div
      class=""
    >
      <ul
        class=""
      >
        <li
          class=""
        >
          <div
            class=""
            role="button"
            style="text-align: left;"
            tabindex="0"
          >
            _defaultPageTitle_summary__resx
          </div>
        </li>
        <li
          class=""
        >
          <div
            class=""
          >
            <div
              role="button"
              tabindex="0"
            >
              <div
                class=""
                style="text-align: left;"
              >
                <span>
                  _pageManagement_postRegSectionHeader__resx
                </span>
                <span
                  class="container"
                >
                  <span
                    aria-hidden="true"
                    class=""
                    role="img"
                  />
                </span>
              </div>
            </div>
            <span
              class="mockNucleusCoreTransition"
            />
          </div>
        </li>
      </ul>
    </div>
    <div
      class=""
    >
      <div
        class=""
      >
        <div
          class=""
        >
          <button
            class=""
            type="button"
          >
            <span
              class=""
            >
              WebsiteNavigatorWidget_LogoutButtonText_default__resx
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WebsiteNavigatorWidget when user is logged in with approval denied status 1`] = `
<div
  role="navigation"
  style="min-height: 1px;"
>
  <a
    class="skipNav"
    href="#main"
    style="border-color: #041532; color: rgb(39, 63, 105); background-color: rgb(247, 247, 247);"
    tabindex="0"
  >
    WebsiteNavigatorWidget_SkipNavigationText_default__resx
  </a>
  <div
    class=""
    style="padding: 15px 15px 15px 15px; background-color: rgb(4, 21, 50); border-radius: 0; flex: 1 1 auto;"
  >
    <div
      class=""
    >
      <ul
        class=""
      >
        <li
          class=""
        >
          <div
            class=""
            role="button"
            style="text-align: left;"
            tabindex="0"
          >
            _defaultPageTitle_summary__resx
          </div>
        </li>
        <li
          class=""
        >
          <div
            class=""
          >
            <div
              role="button"
              tabindex="0"
            >
              <div
                class=""
                style="text-align: left;"
              >
                <span>
                  _pageManagement_postRegSectionHeader__resx
                </span>
                <span
                  class="container"
                >
                  <span
                    aria-hidden="true"
                    class=""
                    role="img"
                  />
                </span>
              </div>
            </div>
            <span
              class="mockNucleusCoreTransition"
            />
          </div>
        </li>
      </ul>
    </div>
    <div
      class=""
    >
      <div
        class=""
      >
        <div
          class=""
        >
          <button
            class=""
            type="button"
          >
            <span
              class=""
            >
              WebsiteNavigatorWidget_LogoutButtonText_default__resx
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WebsiteNavigatorWidget when user is logged in with pending approval status 1`] = `
<div
  role="navigation"
  style="min-height: 1px;"
>
  <a
    class="skipNav"
    href="#main"
    style="border-color: #041532; color: rgb(39, 63, 105); background-color: rgb(247, 247, 247);"
    tabindex="0"
  >
    WebsiteNavigatorWidget_SkipNavigationText_default__resx
  </a>
  <div
    class=""
    style="padding: 15px 15px 15px 15px; background-color: rgb(4, 21, 50); border-radius: 0; flex: 1 1 auto;"
  >
    <div
      class=""
    >
      <ul
        class=""
      >
        <li
          class=""
        >
          <div
            class=""
            role="button"
            style="text-align: left;"
            tabindex="0"
          >
            _defaultPageTitle_summary__resx
          </div>
        </li>
        <li
          class=""
        >
          <div
            class=""
          >
            <div
              role="button"
              tabindex="0"
            >
              <div
                class=""
                style="text-align: left;"
              >
                <span>
                  _pageManagement_postRegSectionHeader__resx
                </span>
                <span
                  class="container"
                >
                  <span
                    aria-hidden="true"
                    class=""
                    role="img"
                  />
                </span>
              </div>
            </div>
            <span
              class="mockNucleusCoreTransition"
            />
          </div>
        </li>
      </ul>
    </div>
    <div
      class=""
    >
      <div
        class=""
      >
        <div
          class=""
        >
          <button
            class=""
            type="button"
          >
            <span
              class=""
            >
              WebsiteNavigatorWidget_LogoutButtonText_default__resx
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WebsiteNavigatorWidget when user is not logged in it renders 1`] = `
<div
  role="navigation"
  style="min-height: 1px;"
>
  <a
    class="skipNav"
    href="#main"
    style="border-color: #041532; color: rgb(39, 63, 105); background-color: rgb(247, 247, 247);"
    tabindex="0"
  >
    WebsiteNavigatorWidget_SkipNavigationText_default__resx
  </a>
  <div
    class=""
    style="padding: 15px 15px 15px 15px; background-color: rgb(4, 21, 50); border-radius: 0; flex: 1 1 auto;"
  >
    <div
      class=""
    >
      <ul
        class=""
      >
        <li
          class=""
        >
          <div
            class=""
            role="button"
            style="text-align: left;"
            tabindex="0"
          >
            _defaultPageTitle_summary__resx
          </div>
        </li>
      </ul>
    </div>
    <div
      class=""
    >
      <div
        class=""
      >
        <div
          class=""
        >
          <button
            class="disabled"
            disabled=""
            type="button"
          >
            <span
              class=""
            >
              _registerNowWidget_textConfig__resx
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
